packages/angular-workspace/example-client-app/src/app/customapp/customapp.component.ts

```diff
@@ -217,12 +224,38 @@ export class CustomAppComponent implements AfterViewInit {
                 statusCode: (tableData.length % 2 === 0) ? 100 : 101,
                 result: possibleStatuses[tableData.length % 3],
                 number: tableData.length / 10,
-                duration: tableData.length * 1000 * (1.1 + 2 * 60 + 3 * 3600)
+                duration: tableData.length * 1000 * (1.1 + 2 * 60 + 3 * 3600),
+                color: this.possibleColors[tableData.length % 5]
             });
         }
         this.tableDataSubject.next(tableData);
     }
 
+    public onMenuButtonColumnBeforeToggle(event: Event): void {
+        const customEvent = event as CustomEvent<MenuButtonColumnToggleEventDetail>;
+        if (customEvent.detail.newState) {
+            this.openMenuButtonColumnRecordId = customEvent.detail.recordId;
+
+            const tableData = this.tableDataSubject.value;
+            const currentRecord = tableData.find(record => record.id === customEvent.detail.recordId)!;
+            this.currentColor = currentRecord.color;
+        }
+    }
+
+    public onColorSelected(color: Color): void {
+        if (!this.openMenuButtonColumnRecordId) {
+            return;
+        }
+
+        const updatedData = this.tableDataSubject.value.map(record => {
+            if (record.id === this.openMenuButtonColumnRecordId) {
+                return { ...record, color };
+            }
+            return record;
+        });
```

Comment:
Note: I tried to write the code more simply, but I ran into and created #2265. I don't think this code is awful, but ideally, I could write it as:

```ts
const record = this.tableDataSubject.value.find(x => x.id === this.openMenuButtonColumnRecordId);
record.color = color;
this.tableDataSubject.next(this.tableDataSubject.value);
```